src/main/java/edu/rpi/legup/puzzle/nurikabe/rules/SurroundRegionDirectRule.java

```diff
@@ -44,12 +52,51 @@ public String checkRuleRawAt(TreeTransition transition, PuzzleElement puzzleElem
         NurikabeCell modCell = (NurikabeCell) modified.getPuzzleElement(puzzleElement);
         modCell.setData(NurikabeType.WHITE.toValue());
 
-        if (contraRule.checkContradiction(modified) == null) {
-            return null;
-        }
-        else {
-            return "Does not follow from this rule at this index";
+        if(cell.getType() == NurikabeType.BLACK) {
+            DisjointSets<NurikabeCell> regions = NurikabeUtilities.getNurikabeRegions(destBoardState);
+            Set<NurikabeCell> adj = new HashSet<>(); //set to hold adjacent cells
+            Point loc = cell.getLocation(); //position of placed cell
+            NurikabeCell cellLeft = destBoardState.getCell(loc.x - 1, loc.y);

```

Comment:
While this is logically correct, this code block uses duplicate code for each direction. I recommend iteration through each direction in a for loop rather than copy pasting the code four times.

Some pseudocode that might help:
```java
import java.awt.Point;

List<Point> directions = { new Point(-1, 0), new Point(1, 0), new Point(0, -1), new Point(0, 1) };

for (Point direction : directions) {
    NurikabeCell cell = destBoardState.getCell(loc.x + direction.x, loc.y + direction.y);
    
    // avoid redundant nested clauses
    if (cell != null && (...) {
        // apply logic
    }
}
```